Source/Testably.Abstractions.Testing/Storage/InMemoryStorage.cs

```diff
@@ -1035,23 +1047,59 @@ private bool IncludeItemInEnumeration(
 			}
 
 			nextLocation = _fileSystem.Storage.GetLocation(container.LinkTarget);
-			if (!_containers.TryGetValue(nextLocation,
-				out IStorageContainer? nextContainer))
+
+			if (!_containers.TryGetValue(nextLocation, out IStorageContainer? nextContainer))
 			{
 				return nextLocation;
 			}
 
+			ThrowOnLinkTypeChange(container, originalLocation, nextContainer);
+
 			container = nextContainer;
 		}
 
 		if (container.LinkTarget != null)
 		{
-			throw ExceptionFactory.FileNameCannotBeResolved(
-				originalLocation.FullPath);
+			throw ExceptionFactory.FileNameCannotBeResolved(originalLocation.FullPath);
```

Comment:
```suggestion
			throw ExceptionFactory.FileNameCannotBeResolved(originalLocation.FullPath, _fileSystem.Execute.IsWindows ? -2147022975 : -2146232800);
```

In the `FileSystemInfoMock` I explicitely set the HResult depending on the mocked OS. Don't you need this here as well to make the tests succeed?

Reply:
Yes, I've it in a unpushed commit, in case I needed to revert the pushed change

Reply:
Fix pushed